src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -197,81 +194,79 @@ class DataModelInstanceRelation(config: RelationConfig, modelExternalId: String)
           (index, field.name, propertyType)
       }
 
-    def parseRow(row: Row): DataModelInstance = {
+    def parseRow(row: Row): DataModelInstanceCreate = {
       val _ = row.get(externalIdIndex) match {
         case x: String => x
         case _ => throw SparkSchemaHelperRuntime.badRowError(row, "externalId", "String", "")
       }
-      val propertyValues: Map[String, Json] = properties.map {
-        case (index, name, propT) =>
-          name -> tryGetValue(propT._1, propT._2).applyOrElse(
-            row.get(index),
-            (_: Any) =>
-              throw SparkSchemaHelperRuntime
-                .badRowError(row, name, propT._1, "")
-          )
-      }.toMap
-      DataModelInstance(modelExternalId, properties = Some(propertyValues))
+      val propertyValues: Map[String, PropertyType] = properties
+        .map {
+          case (index, name, propT) =>
```

Comment:
Could some of this code be nicer if we made a case class for `propertyTypes`? Maybe there's already one we can use!
`val propertyTypes: Map[String, (String, Boolean)]`

Reply:
yes sure, we have a type from scala-sdk which is nice